src/linux/maps_reader.rs

```diff
@@ -130,6 +205,187 @@ impl MappingInfo {
+                debugger_rendezvous.r_version,
+            ));
+        }
+
+        Self::read_link_map(&mut memory_reader, debugger_rendezvous.r_map)
+            .map_err(|e| E::IterateLinkMapFailed(Box::new(e)))
+    }
+
+    fn read_link_map(
+        memory_reader: &mut MemReader,
+        link_map_address: usize,
+    ) -> std::result::Result<Vec<MappingInfo>, FromDebuggerRendezvousError> {
+        use FromDebuggerRendezvousError as E;
+
+        let mut link: link_map = memory_reader
+            .read_pod(link_map_address)
+            .map_err(E::ReadLinkMapEntryFailed)?;
+
+        if link.l_prev != 0 {
+            return Err(E::InvalidLinkEntry);
+        }
+
+        let mut result: Vec<MappingInfo> = Vec::new();
+
+        loop {
+            let name = {
+                let mut buf = Vec::new();
+                memory_reader
+                    .read_until(link.l_name, 0, &mut buf)
+                    .map_err(E::ReadModuleNameFailed)?;
+                buf.pop();
+                OsString::from(String::from_utf8(buf).unwrap())
+            };
+
+            let module_elf_header_address = link.l_addr;
+            let module_elf_header: ElfHeader = memory_reader
+                .read_pod(module_elf_header_address)
+                .map_err(E::ReadElfHeaderFailed)?;
+            validate_elf_signature(&module_elf_header)?;
+
+            let module_program_header_virtual_address =
+                usize::try_from(module_elf_header.e_phoff).unwrap();
+            let module_program_header_address =
+                module_elf_header_address + module_program_header_virtual_address;
+            let module_program_header_len = usize::from(module_elf_header.e_phnum);
+            let module_program_headers: Vec<ProgramHeader> = memory_reader
+                .read_pod_vec(module_program_header_address, module_program_header_len)
+                .map_err(E::ReadProgramHeaderTableFailed)?;
+
+            for module_program_header in module_program_headers
+                .iter()
+                .filter(|x| x.p_type == PT_LOAD)
+            {
+                let segment_virtual_address =
+                    usize::try_from(module_program_header.p_vaddr).unwrap();
+                let segment_address = module_elf_header_address + segment_virtual_address;
+                let segment_len = usize::try_from(module_program_header.p_memsz).unwrap();
+                let segment_end_address = segment_address + segment_len;
+                let segment_offset = usize::try_from(module_program_header.p_offset).unwrap();
+
+                // Round each of these down or up to the nearest page
+                let segment_address = segment_address / 4096 * 4096;
+                let segment_offset = segment_offset / 4096 * 4096;
+                let segment_end_address = segment_end_address.div_ceil(4096) * 4096;
+                let segment_len = segment_end_address - segment_address;
```

Comment:
Is the page size guaranteed to always be 4KiB? It'd be better to use a constant for it and explain why it can't change.